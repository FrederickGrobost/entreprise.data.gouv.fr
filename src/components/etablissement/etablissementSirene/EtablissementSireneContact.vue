<template>
  <div class="company__panel panel">
    <h4>Contact</h4>
    <div class="company__item">
      <label class="company__item-key">Gérant</label>
      <div class="company__item-value">
        {{
          concatNames(resultUniteLegale.prenom_1, resultUniteLegale.nom)
            | ifExist
        }}
      </div>
    </div>
    <div class="company__item">
      <label class="company__item-key">Adresse</label>
      <div class="company__item-value">
        {{ resultSirene.geo_l4 | ifExist }}
      </div>
    </div>
    <div class="company__item">
      <label class="company__item-key">Ville</label>
      <div class="company__item-value">
        {{ resultSirene.code_postal }} {{ resultSirene.libelle_commune }}
      </div>
    </div>
    <div class="company__item">
      <label class="company__item-key">Cedex</label>
      <div class="company__item-value">
        {{ resultSirene.code_cedex | ifExist }}
      </div>
    </div>
    <div class="company__item">
      <label class="company__item-key">Date de création</label>
      <div class="company__item-value">{{ formattedDate }}</div>
    </div>
    <div class="company__item">
      <label class="company__item-key">Tranche d’effectif salariés</label>
      <div class="company__item-value">
        {{ resultSirene.tranche_effectifs | ifExist }}
      </div>
    </div>
  </div>
</template>

<script>
import Filters from "@/components/mixins/filters";
import Formating from "@/components/mixins/formating";

export default {
  name: "EtablissementSireneContact",
  mixins: [Filters, Formating],
  computed: {
    resultSirene() {
      return this.$store.getters.singlePageEtablissementSirene;
    },
    resultUniteLegale() {
      if (this.resultSirene) {
        return this.resultSirene.unite_legale;
      }
      return null;
    },
    formattedDate() {
      if (!this.resultUniteLegale) {
        return null;
      }
      const year = this.resultUniteLegale.date_creation.substring(0, 4);
      const month = this.resultUniteLegale.date_creation.substring(5, 7);
      const day = this.resultUniteLegale.date_creation.substring(8, 10);
      return `${day}/${month}/${year}`;
    }
  }
};
</script>
