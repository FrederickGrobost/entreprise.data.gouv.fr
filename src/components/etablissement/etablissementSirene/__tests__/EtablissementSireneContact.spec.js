import EtablissementSireneContact from "@/components/etablissement/etablissementSirene/EtablissementSireneContact.vue";
import { createLocalVue, mount } from "@vue/test-utils";
import { __createMocks as createStoreMocks } from "@/store/index.js";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);
jest.mock("@/store/index.js");

describe("EtablissementSireneContact.vue", () => {
  test("Computed value formattedDate returns null if no date is present", () => {
    // TODO
  });

  test("Computed value formattedDate compute the correct date if present", () => {
    // TODO
  });

  test("it should compute the full name correctly if name and surname are present", () => {
    // TODO
  });

  test("it should display the name if only the name is present", () => {
    // TODO
  });

  test("it shouldnt display the name if only the first name is present", () => {
    // TODO
  });
});

describe("EtablissementSireneContact.vue snapshot", () => {
  const localVue = createLocalVue();
  const storeMocks = createStoreMocks();
  const wrapperEPC = mount(EtablissementSireneContact, {
    localVue,
    store: storeMocks.store,
    computed: {
      resultSirene() {
        return {
          geo_l4: "52 avenue de la fausse adresse",
          code_postal: "75000",
          libelle_commune: "PARIS",
          code_cedex: "5",
          tranche_effectif: "100",
          unite_legale: {
            prenom_1: "JEAN YVES",
            nom: "SMITH",
            date_creation: "2007-01-01"
          }
        };
      }
    }
  });

  test("It match the snapshot", () => {
    expect(wrapperEPC.vm.$el).toMatchSnapshot();
  });
});
