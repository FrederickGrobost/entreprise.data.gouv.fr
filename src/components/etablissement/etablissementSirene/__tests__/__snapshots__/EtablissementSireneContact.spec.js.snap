// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EtablissementSireneContact.vue snapshot It match the snapshot 1`] = `
<div
  class="company__panel panel"
>
  <h4>
    Contact
  </h4>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Gérant
    </label>
     
    <div
      class="company__item-value"
    >
      
      Jean yves SMITH
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Adresse
    </label>
     
    <div
      class="company__item-value"
    >
      
        
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Ville
    </label>
     
    <div
      class="company__item-value"
    >
      
      75000 PARIS
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Cedex
    </label>
     
    <div
      class="company__item-value"
    >
      
      5
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Date de création
    </label>
     
    <div
      class="company__item-value"
    >
      01/01/2007
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Tranche d’effectif salariés
    </label>
     
    <div
      class="company__item-value"
    >
      
      Non renseigné
    
    </div>
  </div>
</div>
`;
